Initial Map with One Popup

ajchu28 opened the pull request:
Currently the map displays a single popup. Styling is separated in into mapstyle.css for Yuen to add the popup styling. Further steps include writing a script to parse the incoming JSON poem object and add the poem to the map. Public transit stops are currently being displayed, but need to figure out how to add this [shapefile](https://data.cityofchicago.org/Transportation/CTA-Bus-Routes-Shapefile/d5bx-dr8z) of the cta routes to the map.

ajchu28 committed:
initial map with one hardcoded popup

nbriz commented on frontend/mapPage.html:
```diff
@@ -0,0 +1,100 @@
+                'id': 'places',
+                'type': 'symbol',
+                'source': 'places',
+                'layout': {
+                    'icon-image': '{icon}',
+                    'icon-size': 2.5,
+                    'icon-allow-overlap': true
+                }
+            });
+
+            // When a click event occurs on a feature in the places layer, open a popup at the
+            // location of the feature, with description HTML from its properties.
+            map.on('click', 'places', (e) => {
+                // Copy coordinates array.
+                const coordinates = e.features[0].geometry.coordinates.slice();
+                const description = e.features[0].properties.description;
+
+                // Ensure that if the map is zoomed out such that multiple
+                // copies of the feature are visible, the popup appears
+                // over the copy being pointed to.
+                while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+                    coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+                }
+
+                new mapboxgl.Popup()
+                    .setLngLat(coordinates)
+                    .setHTML(description)
+                    .addTo(map);
+            });
+
+            // Change the cursor to a pointer when the mouse is over the places layer.
+            map.on('mouseenter', 'places', () => {
+                map.getCanvas().style.cursor = 'pointer';
+            });
+
+            // Change it back to a pointer when it leaves.
+            map.on('mouseleave', 'places', () => {
+                map.getCanvas().style.cursor = '';
+            });
+        });
```
let's move this code into it's own file, maybe something like `map-logic.js` (@yruann also has some logic in her project && i told her to move it into a file called `map.js` so i'd suggest a different name to avoid merge conflicts), place that in a `js` folder inside the `frontend` folder so that u can then import it into this html page like:
```html
<script src="js/map-logic.js"></script>
```

once the JS is in it's own file, not only will it make our code more modular (&& thus easier to collaborate) but it will also mean ur editor should be able to lint the file for coding style errors

nbriz reviewed (changes requested):
hey @ajchu28 

this works great, but there are a few process + logistical issues.

u should move ur JavaScript code out of the HTML file && into it's own JS file so we can check that it conforms to our [coding style](https://github.com/net-art-uchicago/cta-file-sharing/wiki/coding-style) (see my comment below). ++ also this PR is coming from ur main branch, don't forget we're using a [feature branch workflow](https://github.com/net-art-uchicago/cta-file-sharing/wiki/feature-branch-workflow) so it should be coming from a feature branch.

also, there's a lot of overlap here between what Ying is working on (obviously since u two are collaboration more directly)... but this poses some merging challenges. also, it seems u deleted a couple files u shouldn't have, the `frontend/index.html` && `frontend/css/styles.css`... this would be easier to clean up if u had made these changes on a feature branch, but since u committed them to `main` this can get a little messy... we can clean this up a number of ways, but given ur overlap w/Ying here's what i would suggest we do (let me know what u think):

1. i've left Ying some [feedback in her PR](https://github.com/net-art-uchicago/cta-file-sharing/pull/23), soon as she addresses those i should be able to merge those changes.

2. once those have been merged to the class repo u can try to `git pull upstream main` && see if things get merged cleanly on ur end (including the re-inclusion of those two files u deleted).

3. two options here, depending on how the previous step goes:
  - if the merge went well, then i'd suggest u integrate ur JS code (which u should still move into it's own separate file as mentioned below) into her `map.html` file. (don't forget to create a new [feature branch](https://github.com/net-art-uchicago/cta-file-sharing/wiki/feature-branch-workflow) after pulling changes from main, but before making ur changes)
  - if that merge didn't go well, then maybe u just delete the repo (making sure to save the JS code u wrote somewhere else temporarily so u don't loose it) && start over. create a new fork, clone it, create a feature branch && then integrate ur JS code as described above.

ajchu28 commented:
Thanks for the detailed feedback! I went ahead and reverted my branch back to the class main branch to recover the files I deleted and switched to a feature branch. I then reorganized my code into a `mapview.html` file, `mapScript.js` file, and moved my css into `styles.css`. I was able to fix most of javascript file to conform to the class standards, and the page loads as normal if I comment out a few lines. The problem lines for me are highlighted in the screenshot below:

<img width="317" alt="Screen Shot 2022-11-24 at 11 24 38 AM" src="https://user-images.githubusercontent.com/84213588/203839667-8e63a8cb-16f1-4f8d-871a-6eafcbe6116d.png">

If I comment this out, the page loads and views the map as before. However, to conform to the code styling, I have to delete the hyphens or the page doesn't load properly. However, the hyphenated attributes are defined by Mapbox API, and the popups won't display if I change the names. The documentation is [here](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#symbol) if you scroll down to the icon-image section. What do you think is the best way to work around this?

nbriz commented:
@ajchu28 thnx for the detailed description of changes! i still have to give a closer look, but sounds like this will be ready to merge soon as u fix the hyphenated issue, try putting single quotes around any property names that have hyphens, like this:

```js
layout: {
  'icon-image': '{icon}',
  'icon-size': 2.5,
  'icon-allow-overlap': true
}
```

nbriz commented:
@ajchu28 i'm going to close this PR now since u opened the other one
